src/caesium/crypto/generichash.clj

```diff
@@ -60,15 +60,17 @@
   ([buf msg]
    (blake2b-to-buf! buf msg {}))
   ([buf msg {:keys [key salt personal] :or {key (bb/alloc 0)}}]
-   (if (or salt personal)
-     ;; You can't set the defaults in the argspec's destructuring form,
-     ;; because you want to be able to differentiate between a salt that
-     ;; wasn't passed and an empty salt, to call a different fn.
-     (let [salt (or salt (bb/alloc blake2b-saltbytes))
-           personal (or personal (bb/alloc blake2b-personalbytes))]
-       (b/call! blake2b-salt-personal buf msg key salt personal))
-     (b/call! blake2b buf msg key))
-   buf))
+   (if-not (zero?
+            (if (or salt personal)
```

Comment:
I'd be a little happier if this was factored out into a let to clarify that it's the result of calling blake2b, I think the nesting makes it tricky to read.

Reply:
If I understood you correctly, I just addressed this in 8acf2b8.